xds/src/main/java/io/grpc/xds/ClusterResolverLoadBalancer.java

```diff
@@ -226,8 +226,8 @@ private void handleEndpointResourceUpdate() {
       }
       if (addresses.isEmpty()) {
         if (endpointNotFound.isOk()) {
-          endpointNotFound =
-              Status.UNAVAILABLE.withDescription("No usable endpoint from any cluster");
+          endpointNotFound = Status.UNAVAILABLE.withDescription(
+              "No usable endpoint from cluster(s): " + clusters);
```

Comment:
Would it be possible to include the health state of cluster/endpoints if available? I have been seeing the "No usable endpoint..." message but no more info to diagnose the problem

Reply:
This has been improved a bit in this PR. For cases that TD sends empty EDS responses or with endpoints being unhealthy, it's a normal xDS operation but our LB policy is not able to proceed with the given data. So we have to make up the error status for failing RPCs with this reason. The new status is slightly better than before, it includes the clusters that the LB policy is currently resolving endpoints for. Regarding including health state of endpoints (if any), it's  actually quite complicated to collect, aggregate and put together that information for all clusters and make it into the overall status. It's doable, but I'd hope to make that change separately, as that change itself may need to move code around quite bit.

For other cases that the LB policy does get error from endpoint resolution operations (e.g., DNS failure, xDS stream failures) and no endpoints can be found. That error status is included in the UNAVAILABLE for failing application RPCs. This works in a similar way as gRPCLB.

Reply:
Yes something is better than nothing. I wasn't even sure if the health information is available based on the received xDS response but based on what you are saying looks like it is available although "... it's actually quite complicated to ... make it into the overall status". Hopefully we are able to do it sometime soon to help our users troubleshoot better.